plenum/server/replicas.py

```diff
@@ -47,20 +46,24 @@ def grow(self) -> int:
                        extra={"tags": ["node-replica"]})
         return self.num_replicas
 
-    def shrink(self) -> int:
-        replica = self._replicas[-1]
-        self._replicas = self._replicas[:-1]
-        self._messages_to_replicas = self._messages_to_replicas[:-1]
-        self._monitor.removeInstance()
+    def remove_replica(self, index: int=None):
+        if index >= self.num_replicas:
+            return
+        if index is None:
+            index = self.num_replicas - 1
+        replica = self._replicas.pop(index)
+        for request_key in replica.requestQueues:
```

Comment:
`requestQueues` is not a request queue, it is a map from ledger IDs to request queues.

Reply:
By default, the iterator of the Dict structure follows the keys. But this is not obvious, so changed to replica.requestQueues.keys()
All tests from test_replica_removing.py check it.

Reply:
Fixed it.
Now all tests from test_replica_removing.py check it.

Reply:
It makes sense to add verification of freeing the request by the replica to `test_replica_removing_in_ordering`.